state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -942,9 +945,12 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 				TransactionOutcome::Commit(Ok(()))
 			},
 			Err(AllBatchError::NotRequired) => TransactionOutcome::Commit(Ok(())),
-			Err(AllBatchError::Other) => TransactionOutcome::Rollback(Err(DispatchError::Other(
-				"AllBatch ApiCall creation failed, rolled back storage",
-			))),
+			Err(error) => {
+				Self::deposit_event(Event::<T, I>::FailedToBuildAllBatchCall { error });
+				TransactionOutcome::Rollback(Err(DispatchError::Other(
+					"AllBatch ApiCall creation failed, rolled back storage",
+				)))
```

Comment:
This event will not be visible because of the storage rollback.

We will have to return the error from `do_egress_scheduled_fetch_transfer` along with the transaction outcome and then emit it after the storage transaction rollback.